### PR TITLE
Add self query translator for weaviate vectorstore

### PR DESCRIPTION
**Pull request by domchan**
# Add self query translator for weaviate vectorstore

Adds support for the EQ comparator and the AND/OR operators. 

@dev2049

### CODE REVIEW

**Comment by dev2049** [comment]
amazing! if you have time it'd be great to add an example notebook (like the ones for chroma and pinecone). if not i can take a stab later today

**Comment by domchan** [comment]
> amazing! if you have time it'd be great to add an example notebook (like the ones for chroma and pinecone). if not i can take a stab later today

no problem I'll put together an example based on the same content as the others

**Comment by domchan** [comment]
> amazing! if you have time it'd be great to add an example notebook (like the ones for chroma and pinecone). if not i can take a stab later today

Done. Largely copied from the Chroma example but has been tested with weaviate.

File: langchain/retrievers/self_query/weaviate.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+"""Logic for converting internal query language to a valid Chroma query."""
```
Comment (hwchase17) [inline langchain/retrievers/self_query/weaviate.py:1]: nit: docstring

**Review by hwchase17 — APPROVED** [review]
largely lgtm!

**Pull request merged**